presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java

```diff
@@ -34,16 +36,42 @@
     private final List<String> bucketedBy;
     private final int bucketCount;
     private final List<SortingColumn> sortedBy;
+    private final Optional<List<Type>> types;
+    private final BucketingType bucketingType;
+
+    public HiveBucketProperty(
+            List<String> bucketedBy,
+            int bucketCount,
+            List<SortingColumn> sortedBy)
+    {
+        this(bucketedBy, bucketCount, sortedBy, Optional.empty());
+    }
 
     @JsonCreator
     public HiveBucketProperty(
             @JsonProperty("bucketedBy") List<String> bucketedBy,
             @JsonProperty("bucketCount") int bucketCount,
-            @JsonProperty("sortedBy") List<SortingColumn> sortedBy)
+            @JsonProperty("sortedBy") List<SortingColumn> sortedBy,
+            @JsonProperty("types") Optional<List<Type>> types)
     {
         this.bucketedBy = ImmutableList.copyOf(requireNonNull(bucketedBy, "bucketedBy is null"));
         this.bucketCount = bucketCount;
         this.sortedBy = ImmutableList.copyOf(requireNonNull(sortedBy, "sortedBy is null"));
+        this.types = requireNonNull(types, "type is null");
+        if (types.isPresent()) {
```

Comment:
Let's set it explicitly as a constructor parameter. Lets check if the `types` is set if the `bucketingType` is prestonative